### PR TITLE
Add .editorconfig

### LINKED ISSUE #22: Create a .editorconfig file
# Website/API Suggestion

## Description

Basic code style settings like line endings, file endings, tab vs spaces & indent size can be set in a .editorconfig file that most IDEs will detect and follow.

## Why

Consistent code style.

### PR DESCRIPTION
**Pull request by donaldducky**
fixes #22

### CODE REVIEW

**Comment by donaldducky** [comment]
Running this tool:
https://github.com/editorconfig-checker/editorconfig-checker.javascript

Produces the following output:
Screen Shot 2019-10-08 at 6 12 29 PM

All of them are easily fixable except for `db/seeds/data/libraries.yml`.
Those lines 108-110 are formatted that way for the python code in there.

I'm not too sure on how to fix that issue atm (aside from ignoring it explicitly).

**Comment by MattIPv4** [comment]
Seed data can be ignored as it’s auto generated — this looks great, I think it’d be awesome to add a commit here that fixes these errors and maybe add this checker to the test suite, though you can do that in a different PR if you’d prefer (or not at all).

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Ignored paths
+[*.{psd,opts}]
+charset = none
+end_of_line = none
+indent_size = none
+indent_style = none
+insert_final_newline = none
+trim_trailing_whitespace = none
+
+[db/seeds/data/libraries.yml]
```
Comment (MattIPv4) [inline .editorconfig]: I would exclude all seed data here, not just this specific file?

Comment (donaldducky) [reply .editorconfig]: Should it be `db/seeds/*` or `db/seeds/data/*`?

Comment (MattIPv4) [reply .editorconfig]: data :)

**Comment by donaldducky** [comment]
For the ci portion, I can add an npm run command: `"lint:editorconfig": "editorconfig-checker"`

In the `.github/workflows/ci.yml` should I just add a step after `npm install`?

I haven't worked with the workflows before but I guess it really depends when/how you want this to fail the build.

If you want to force the build to fail with this check, I'd suggest maybe moving `npm install` to the top and then running this check. If those pass, do the db creation and the rest of the ci.

**Comment by MattIPv4** [comment]
I think the best bet would actually to be to integrate it into the mocha test suite — I realise it’s more work but it means it’ll get checked whenever and where ever tests get run, not just ci.

**Comment by MattIPv4** [comment]
If you want to split that into a separate PR and merge this first, happy to do that too.

**Comment by donaldducky** [comment]
I think it makes sense to include it as a part of this so the editorconfig is enforced as soon as it's merged in.

I'm looking at: `deploy/src/test.js`

Would you want something like this?

```javascript
    // Run the test suite
    try {
        out = await run('npm run lint:editorconfig');
        out += await run('npm run test -- --skip-lighthouse --reporter dot');
    } catch (e) {
        err = e;
    }
```

**Comment by MattIPv4** [comment]
Not there, those only get run when I deploy to staging or production — it’ll want to live in the `/test` directory as a mocha test file. As it isn’t specific to any part of the code base, it can live at the root like the lighthouse test file. Either it can make a call to the npm command as the “test” using exec, or if there’s programmatic integration available for the tool that’d be best.

**Comment by MattIPv4** [comment]
It’ll probably be a super basic mocha file:

describe editorconfig
it passes
run exec or integration
expect status to be okay

**Comment by donaldducky** [comment]
Ok, I'll have a look in a little bit.

I'll add a test file in that folder called editorconfig with the describe block and it'll have to shell out to exec. I don't see a programmatic interface for that tool, unfortunately :(

**Comment by MattIPv4** [comment]
Awesome, sounds like a plan, tyvm.

**Comment by donaldducky** [comment]
@MattIPv4 I think this is good now, have the test in place. It passed in CI.

Here's what the output looks like when it fails:

Screen Shot 2019-10-08 at 8 08 40 PM

Tested it using the following command:
`./node_modules/mocha/bin/mocha test/editorconfig.js`

**Review by MattIPv4 — APPROVED** [review]
Lgtm.

**Comment by donaldducky** [comment]
Cool, totally forgot to remove the `expect` and `secret`.

Curious about the `done => {}` to `function (done) {}` change.
Guess it's just for scoping `this.retries` and `this.slow`?

Would it be preferable to change all the tests to use that syntax rather than the arrow function? Or are you only using the function syntax when you need access to `this`?

**Comment by MattIPv4** [comment]
Yeah, only using `function` where I need access to `this` :)

**Pull request merged**